pkg/fleethttp/fleethttp.go

```diff
@@ -108,3 +112,20 @@ func NewTransport(opts ...TransportOpt) *http.Transport {
 func noFollowRedirect(*http.Request, []*http.Request) error {
 	return http.ErrUseLastResponse
 }
+
+// NewGithubClient returns an HTTP client customized for accessing Github.
+//
+// - If the NETWORK_TEST_GITHUB_TOKEN variable is empty, then this is equivalent to
+// call `NewClient()`.
+// - If the NETWORK_TEST_GITHUB_TOKEN variable is set, then the client will use the
+// token for authentication (as OAuth2 static token).
+func NewGithubClient() *http.Client {
+	if githubToken := os.Getenv("NETWORK_TEST_GITHUB_TOKEN"); githubToken != "" {
```

Comment:
nit: should we consider removing the `NETWORK_TEST` prefix? I understand we're setting it only for tests, but seems like it can be used more generally too?

Reply:
I didn't use `GITHUB_TOKEN` to not mess up any production environments (in case the variable is set for some reason). If we see any users hit this issue then we can consider making the variable "production available" (by removing the `NETWORK_TEST_` prefix).

Let me know if that makes sense.